core/trino-main/src/main/java/io/trino/event/QueryMonitor.java

```diff
@@ -346,6 +355,30 @@ private Optional<String> createTextQueryPlan(QueryInfo queryInfo)
         return Optional.empty();
     }
 
+    private Optional<String> createAnonymizedPlan(QueryInfo queryInfo)
+    {
+        try {
+            if (queryInfo.getOutputStage().isPresent()) {
+                Map<PlanFragmentId, PlanNode> anonymizedPlan = new HashMap<>();
+                AnonymizePlanRewriter rewriter = new AnonymizePlanRewriter(metadata, queryInfo.getSession().toSession(sessionPropertyManager));
+                List<StageInfo> allStages = getAllStages(queryInfo.getOutputStage());
+                for (StageInfo stageInfo : allStages) {
+                    if (stageInfo.getPlan() == null) {
+                        continue;
+                    }
+                    anonymizedPlan.put(stageInfo.getPlan().getId(), rewriteWith(rewriter, stageInfo.getPlan().getRoot()));
+                }
+                if (!anonymizedPlan.isEmpty()) {
+                    return Optional.of(anonymizedPlanCodec.toJson(anonymizedPlan));
```

Comment:
One issue here is that we're relying on AnonymizePlanRewriter to anonymise the PlanNode and then printing the whole PlanNode. It's very easy to miss updating AnonymizePlanRewriter after some new info is added to a PlanNode. 
So maybe we should rather have `AnonymizePlanRewriter` itself produce the anonymized plan fragment string and we can explicitly print the fields that we need there instead of relying on json serialisation of PlanNode. `JoinOrderPrinter` is also an example of similar approach.